src/clusterfuzz/_internal/tests/appengine/libs/issue_filer_test.py

```diff
@@ -366,6 +411,45 @@ def test_testcase_metadata_invalid(self):
         ['ClusterFuzz', 'Reproducible', 'Pri-1', 'Stability-Crash', 'Type-Bug'],
         issue_tracker._itm.last_issue.labels)
 
+  def test_testcase_save_exception(self):
+    """Tests issue filing when issue.save exception"""
+    self.mock.get.return_value = CHROMIUM_POLICY_FALLBACK
+    original_save = monorail.issue.Issue.save
+    helpers.patch(self, ['libs.issue_management.monorail.issue.Issue.save'])
+
+    def my_save(*args, **kwargs):
+      if getattr(my_save, 'raise_exception', True):
+        setattr(my_save, 'raise_exception', False)
+        raise Exception("Boom!")
+      else:
+        return original_save(*args, **kwargs)
+
+    self.mock.save.side_effect = my_save
+
+    issue_tracker = monorail.IssueTracker(IssueTrackerManager('chromium'))
+    issue_filer.file_issue(self.testcase5, issue_tracker)
+
+    six.assertCountEqual(self,
+                         ['fallback>component', '-component1', '-component2'],
+                         issue_tracker._itm.last_issue.components)
+    self.assertTrue(
+        issue_tracker._itm.last_issue.body.find('**NOTE**: This bug was') != -1)
```

Comment:
nit: use `self.assertIn`. 

Also let's test that the substitution worked as well.

Reply:
> Also let's test that the substitution worked as well.

This is testing it:

```python
    self.assertTrue(
        issue_tracker._itm.last_issue.body.endswith(
            ' '.join(metadata_components)))
```

Do you mean something like this instead?

```python
    self.assertTrue(
        issue_tracker._itm.last_issue.body.endswith(
            'component1 component2')))
```

Reply:
Ah I missed that. What I meant was:

```python
self.assertIn('**NOTE**: This bug was .. (the full message)...  component1 component2)', last_issue.body)
```

To be more explicit.

Reply:
Done.